### PR TITLE
CON-2433: refactored activeFilters logic

### PR DESCRIPTION
**Pull request by kernelkiller**

### CODE REVIEW

File: components/FilterBar/selectors.js [hunk 1]
```diff
@@ -3,33 +3,42 @@ import {
...
 );
 
 /**
- * Selects the summed up shipping costs of the cart.
- * @type {number}
+ * Checks if the filter bar is shown within a category page
+ * @return {bool}
  */
-export const isFilterBarShown = createSelector(
+export const isFilterBarShownInCategory = createSelector(
```
Comment (devbucket) [inline components/FilterBar/selectors.js]: If the selector is specific like that then it should belong to the Category Page, not the FilterBar anymore. And THEN you should rename it back to `isFilterBarShown`.

File: components/FilterBar/selectors.js [hunk 2]
```diff
@@ -3,33 +3,42 @@ import {
...
-    return false;
+    return !subCategories && (hasProducts || activeFilters);
   }
 );
+
+/**
+ * Checks if the filter bar is shown within a search page
+ * @return {bool}
+ */
+export const isFilterBarShownInSearch = createSelector(
```
Comment (devbucket) [inline components/FilterBar/selectors.js]: If the selector is specific like that then it should belong to the Search Page, not the FilterBar anymore. And THEN you should rename it back to `isFilterBarShown`.

File: components/FilterBar/selectors.js (hunk 1 above)
Comment (kernelkiller) [reply components/FilterBar/selectors.js]: moved to a custom selector

File: components/FilterBar/selectors.js (hunk 2 above)
Comment (kernelkiller) [reply components/FilterBar/selectors.js]: moved to a custom selector

**Pull request merged**